### PR TITLE
Migrate to capacitor v4

### PR DESCRIPTION
**Pull request by MateiNenciu**

### CODE REVIEW

**Comment by eaglefsd** [comment]
@RaphaelWoude any chance to merge and publish this? :)
We need to update capacitor in our project in order to install another plugin, which, unfortunately, isn't compatible with capacitor 3.

File: ios/Podfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-platform :ios, '12.0'
+platform :ios, '13.0'
 
 def capacitor_pods
   # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
```
Comment (RaphaelWoude) [inline ios/Podfile:4]: https://capacitorjs.com/docs/updating/4-0#raise-ios-deployment-target

The require relative needs to be present for v4.
Post install needs to be present for v4.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "capacitor-native-settings",
-  "version": "0.2.1",
+  "version": "4.0.0",
```
Comment (RaphaelWoude) [inline package.json:3]: 0.2.1 to 4.0.0 seems like a very big jump.
Let's set the version to 0.3.0

**Comment by RaphaelWoude** [comment]
Thank you for your contribution. Please look at the comments I made on your PR.
After the comments have been resolved, I will merge the PR & add the new version to NPM.

**Review by RaphaelWoude — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (MateiNenciu) [reply package.json:3]: > 0.2.1 to 4.0.0 seems like a very big jump. Let's set the version to 0.3.0

I followed the convention from other capacitor community plugins, they all jumped to v4 to better show compatibility with capacitor 4, and not 3

File: ios/Podfile (hunk 1 above)
Comment (MateiNenciu) [reply ios/Podfile:4]: > https://capacitorjs.com/docs/updating/4-0#raise-ios-deployment-target
> 
> The require relative needs to be present for v4. Post install needs to be present for v4.

I believe those changes for Podfile are meant only for the main project. I checked other capacitor v4 plugins migrations and they didn't include those changes. Take for example: https://github.com/capawesome-team/capacitor-firebase/pull/168/files

File: package.json (hunk 2 above)
Comment (MateiNenciu) [reply package.json:3]: If the jump is too big to v4 I strongly recommend at least to raise version to 1.0.0 because it has breaking changes

Comment (RaphaelWoude) [reply package.json:3]: > I followed the convention from other capacitor community plugins, they all jumped to v4 to better show compatibility with capacitor 4, and not 3

Ah I see. Lets keep it at **v4.0.0**

File: ios/Podfile (hunk 1 above)
Comment (RaphaelWoude) [reply ios/Podfile:4]: Ah my bad. It does seem that this is only for main projects.

**Review by RaphaelWoude — APPROVED** [review]

**Comment by RaphaelWoude** [comment]
Live in v4.0.0

**Pull request merged**